README.md

```diff
@@ -1,15 +1,120 @@
-# Igor
-[![Build Status](https://api.travis-ci.org/spinnaker/igor.svg?branch=master)](https://travis-ci.org/spinnaker/igor)
+Igor is a service that provides a single point of integration with CI and SCM services for Spinnaker.
 
-Igor provides a single point of integration with Jenkins, Travis and Git repositories ( BitBucket, Stash, and Github ) within Spinnaker.
+# Common Polling Architecture
+
+Igor runs a number of pollers that all share the same common architecture. At a high level, they all:
+
+- periodically get a list of items from an external resource (e.g. builds on a Jenkins master)
+- compare that list against their own persisted cache of items (the difference is called _delta size_)
+- send an `echo` event for each new item
+- caches the new list of items
+
+Features:
+
+- *locking*: pollers can optionally acquire a distributed lock in the storage system before attempting to complete a polling cycle. This makes it possible to run Igor in a high-availability configuration and scale it horizontally.
+- *safe guards*: abnormally large delta sizes can indicate a problem (e.g. lost or corrupt cache data) and cause downstream issues. If a polling cycle results in a delta size above the threshold
+- *health*: Igor has a `HealthIndicator` that reports `Down` if no pollers are running or if they have not had a successful polling cycle in a long time
+
+Relevant properties:
+
+| *Property* | *Default value* | *Description* |
+| --- | --- | --- |
+| `spinnaker.build.pollInterval` | `60` | Interval in seconds between polling cycles |
+| `spinnaker.pollingSafeguard.itemUpperThreshold` | `1000` | Defines the upper threshold for number of new items before a cache update cycle will be rejected | `locking.enabled` | `false` | Enables distributed locking so that Igor can run on multiple nodes without interference |
+
+Relevant metrics:
+
+| *Property* | *Type* | *Description* |
+| --- | --- | --- |
+| `pollingMonitor.newItems` | gauge | represents the number of new items cached by a given monitor during a polling cycle |
+| `pollingMonitor.itemsOverThreshold` | gauge | 0 if deltaSize < threshold, deltaSize otherwise |
+| `pollingMonitor.pollTiming` | timer | published for every polling cycle with the duration it took to complete |
```

Comment:
do you think it's worth mentioning this event is fired with the name of the monitor as a tag `monitor`